massgov/mayflower
# DP-17156 Replace uglify-js with terser

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)


## Description

uglify-es is no longer maintained and uglify-js does not support ES6+.
terser is a fork of uglify-es that mostly retains API and CLI compatibility with uglify-es and uglify-js@3.

1.  Remove uglify.
2.  Install terser.
3.  Update patternlab/styleguide/tools/gulp/pipelines/js.js with terser.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-17156)
- [Github issue](https://github.com/massgov/mayflower/pull/875)

## Steps to Test


1.  Build Patternlab with `npm i & gulp`.
2. Watch the log for any errors.
3. Confirm the build is successful with no error.

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


* 

#### @TODO


*

#### Today I learned...
